fahrschein/src/main/java/org/zalando/fahrschein/ReaderManager.java

```diff
@@ -0,0 +1,10 @@
+package org.zalando.fahrschein;
+
+import org.zalando.fahrschein.domain.Subscription;
+
+import java.util.Optional;
+
+@FunctionalInterface
+public interface ReaderManager {
```

Comment:
I'd prefer the separate interface. The current signature only works for the subscription api, to also support the lowlevel api the signature would need to be

    boolean continueReading(Set<String> eventNames);

Or

    boolean continueReading(Set<String> eventNames, Optional<Subscription> subscription);